src/server/transaction.cc

```diff
@@ -635,27 +635,26 @@ void Transaction::ScheduleInternal() {
     txid_ = op_seq.fetch_add(1, memory_order_relaxed);
     time_now_ms_ = GetCurrentTimeMs();
 
-    atomic_uint32_t lock_granted_cnt{0};
-    atomic_uint32_t success{0};
-
-    auto cb = [&](EngineShard* shard) {
-      auto [is_success, is_granted] = ScheduleInShard(shard);
-      success.fetch_add(is_success, memory_order_relaxed);
-      lock_granted_cnt.fetch_add(is_granted, memory_order_relaxed);
+    atomic_uint32_t schedule_fails = 0;
+    auto cb = [this, &schedule_fails](EngineShard* shard) {
+      if (!ScheduleInShard(shard)) {
+        schedule_fails.fetch_add(1, memory_order_relaxed);
+      }
     };
     shard_set->RunBriefInParallel(std::move(cb), is_active);
 
-    if (success.load(memory_order_acquire) == unique_shard_cnt_) {
+    if (schedule_fails.load(memory_order_relaxed) == 0) {
       coordinator_state_ |= COORD_SCHED;
-      if (lock_granted_cnt.load(memory_order_relaxed) == unique_shard_cnt_) {
-        coordinator_state_ |= COORD_OOO;  // If we granted all locks, we can run out of order.
-      }
+
+      bool all_ooo = false;
```

Comment:
what's the reason for removing `lock_granted_cnt` and iterating again here?

Reply:
1. We iterated before over the shard data to set OOO flags
2. Why duplicate the state
3. Reduce number of atomic ops, now we have 0 on average, before we had 2*N

Reply:
Yes, that's my question from above about stats. But we can't easily compare this value to our current stats (for example to get a percentage breakdown of transaction types) because it's sum is multiplied by the width (we can take the average width to convert it back  🤔 )

Reply:
we can not because OOO transaction is not defined well now as a term. So we will remove this from the stats.
Instead we will have tx_shard_executions vs tx_shard_ooo_executions or something like this